app.rb

```diff
@@ -0,0 +1,158 @@
+require_relative 'teacher'
+
+class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_all_rentals,
+    '7' => :exit_program
+  }.freeze
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start
+    loop { choose_option }
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts "There are #{@books.count} book(s) available."
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(age, name)
+    when '2'
+      create_teacher(age, name)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(age, name)
+    print 'Grade: '
+    grade = gets.chomp
+    print 'Parent permission? [Y/N]: '
+    permission_option = gets.chomp.downcase
+    student = Student.new(grade, age, name, parent_permission: permission_option == 'y')
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
+
+  def create_teacher(age, name)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    if title.strip != '' && author.strip != ''
+      book = Book.new(title, author)
+      @books << book
+      puts 'Book created successfully.'
+    else
+      puts 'Please enter the book title and author.'
+    end
+  end
+
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'Nothing to see here.'
+    else
+      puts 'Enter the number of the book you want: '
+      list_books
+      book_number = gets.chomp.to_i - 1
+      puts 'Enter your ID: '
+      list_people
+      person_id = gets.chomp.to_i
+      individual = @people.find { |person| person.id == person_id }
+
+      print 'Enter the date [yyyy-mm-dd]: '
+      date = gets.chomp.to_s
+      rental = Rental.new(date, @books[book_number], individual)
+      @rentals << rental
+      puts 'Book rented successfully.'
+    end
+  end
```

Comment:
Example video
![image](https://github.com/melkamu12/OOP-school-library/assets/78251927/460e8a47-bade-4ab5-a92e-450e787df82a)

Your app
![image](https://github.com/melkamu12/OOP-school-library/assets/78251927/463396e8-c182-4fcb-a3ae-aa82ecd9a867)

In the example video, please note that a rental is created by selecting a person in the list by number (not id) 🤗

I tried selecting a person using number but the app crashes, please correct that so that the console app works as expected. 🛠